asdfghjkl/vector.py

```diff
@@ -0,0 +1,131 @@
+import copy
+from typing import List, Dict
+
+import torch
+import torch.nn as nn
+import torch.distributed as dist
+
+__all__ = ['ParamVector', 'reduce_vectors', 'normalization', 'orthnormal']
+
+
+class ParamVector:
+    def __init__(self, params: List[torch.Tensor], vectors):
+        assert len(params) > 0, 'params cannot be empty.'
+
+        self.params: List[torch.Tensor] = params
+        self.vectors: Dict[torch.Tensor, torch.Tensor] = {}
+
+        if isinstance(vectors, torch.Tensor):
+            assert vectors.ndim == 1
+            pointer = 0
+            for p in params:
+                numel = p.numel()
+                v = vectors[pointer: pointer + numel]
+                self.vectors[p] = v.view_as(p)
+                pointer += numel
+            assert pointer == vectors.numel()
+        elif isinstance(vectors, list):
+            for p, v in zip(params, vectors):
+                assert p.shape == v.shape
+                self.vectors[p] = v
+        elif isinstance(vectors, dict):
+            for p in params:
+                assert p.shape == vectors[p].shape
+            self.vectors = vectors
+        else:
+            raise TypeError(f'Invalid vectors type: {type(vectors)}')
+
+    def values(self):
+        return self.vectors.values()
```

Comment:
Here, `values()` should be guaranteed to have the same order as inserting. I think it might be wrong if python<3.6. https://stackoverflow.com/questions/34305003/difference-between-dictionary-and-ordereddict. So, personally, I prefer to use `OrderedDict`

Reply:
Thanks, that's a good point! I will fix it to use `OrderedDict`

Reply:
done!